compiler/rustc_resolve/src/imports.rs

```diff
@@ -546,6 +555,41 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
             self.finalize_resolutions_in(*module);
         }
 
+        for (_module_def_id, module_children) in
+            self.tcx.with_stable_hashing_context(|hcx| self.module_children.to_sorted(&hcx, false))
+        {
+            for import in module_children {
+                if !import.vis.is_public() {
```

Comment:
It's better to move the lint logic to `check_hidden_glob_reexports`.

That function already has a very similar logic for iterating over bindings.
And also the effective visibility table is already populated there so you can do checks like `self.effective_visibilities.is_exported(def_id)` instead of just relying on `pub` and getting false positives.